dm/syncer/syncer.go

```diff
@@ -2290,16 +2274,50 @@ func (s *Syncer) Run(ctx context.Context) (err error) {
 					s.tctx.L().Info("re-replicate shard group was completed", zap.String("event", "XID"), zap.Stringer("re-shard", shardingReSync))
 					err = closeShardingResync()
 					if err != nil {
-						return terror.Annotatef(err, "shard group current location %s", shardingReSync.currLocation)
+						return false, terror.Annotatef(err, "shard group current location %s", shardingReSync.currLocation)
+
 					}
-					continue
+					return true, nil
 				}
 			}
 
 			s.tctx.L().Debug("", zap.String("event", "XID"), zap.Stringer("last location", lastTxnEndLocation), log.WrapStringerField("location", endLocation))
 
 			job := newXIDJob(endLocation, startLocation, endLocation)
-			_, err2 = s.handleJobFunc(job)
+			_, err = s.handleJobFunc(job)
+			return needContinue, err
```

Comment:
```suggestion
			return false, err
```

use explicit value for better understanding